kong/plugins/aws-lambda/handler.lua

```diff
@@ -250,7 +250,13 @@ function AWSLambdaHandler:access(conf)
   -- instead of JSON arrays for empty arrays.
   if conf.empty_arrays_mode == "legacy" then
     local ct = headers["Content-Type"]
-    if ct and ct:lower():match("application/.*json") then
+    -- If Content-Type is specified by multiValueHeader then
+    -- it will be an array, so we need to get the first element
+    if type(ct) == "table" and #ct > 0 then
+      ct = ct[1]
+    end
+
+    if ct and type(ct) == "string" and ct:lower():match("application/.*json") then
```

Comment:
here `"application/.*json"` maybe cannot JIT, I remember that the JIT of the match is switched, and the RE match cannot be JIT.

Reply:
Good point, I moved to `ngx.re`

Reply:
sorry, maybe this: https://web.archive.org/web/20220717120825/http://wiki.luajit.org/NYI

> LuaJIT 2.1 adds a trace stitching feature which allows traces to stop at a classic C function or a not-compiled built-in, return to the interpreter, run the C function or built-in and then start a new trace after it returns. This is not particularly efficient, but it avoids trace aborts due to NYI functions, which would previously force the whole code path around such a function to be interpreted.

I understand it is optimized from NYI to stitches in 2.1, but it not be JIT

Reply:
Yes that's also what I read through, as long as it is marked as `stitch` I think then it cannot be fully compiled